lib/oanda_api/client/client.rb

```diff
@@ -158,9 +170,11 @@ def self.last_throttled_at
     # @param [Time] time The time that the throttle was requested.
     #
     # @return [void]
-    def self._throttle(time)
-      @throttle_mutex.synchronize { @throttled_at = time }
-      sleep OandaAPI.configuration.min_request_interval
+    def self._throttle(delta, time)
+      @throttle_mutex.synchronize do
+        @throttled_at = time
+        sleep OandaAPI.configuration.min_request_interval - delta
```

Comment:
After thinking it through again, I see that you're absolutely right. Great catch! The current logic allows multiple threads to all pass through the check that asks whether or not the minimum request interval has been exceeded, and then on waking permits them all to carry on.